nncf/quantization/algorithms/weight_compression/algorithm.py

```diff
@@ -173,58 +260,129 @@ def apply(
+        self,
+        model: TModel,
+        graph: NNCFGraph,
+        nodes_to_compress: List[NNCFNode],
+        activations: Optional[Dict[str, List[Tensor]]] = None,
+    ) -> TModel:
+        all_weight_params: List[WeightCompressionParameters] = []
+        weight_names = set()
+
+        is_last_layer_shared = False
+        n = len(nodes_to_compress)
+        for i, node in enumerate(nodes_to_compress):
+            for weight_name, weight_port_id in self._backend_entity.get_weight_names_and_port_ids(node, graph):
+                if weight_name in weight_names:
+                    if i == n - 1:
+                        is_last_layer_shared = True
+                    continue
+
+                weight = self._backend_entity.get_weight(node, weight_port_id, model, graph)
+                if weight.dtype not in [TensorDataType.float32, TensorDataType.float16, TensorDataType.float64]:
+                    continue
+                reduction_axes = self._backend_entity.get_channel_agnostic_reduction_axes(node, weight_port_id, graph)
+                if isinstance(reduction_axes, tuple) and len(reduction_axes) != 1:
+                    nncf_logger.warning(
+                        f"Weight compression expects a single reduction axis, but {len(reduction_axes)} given. "
+                        f"Weight shape: {weight.shape}, reduction axes: {reduction_axes}, "
+                        f"node name: {node.node_name}. The node won't be quantized."
+                    )
+                    continue
+                reduction_axis = reduction_axes[0] if isinstance(reduction_axes, tuple) else reduction_axes
+
+                weight_params = WeightCompressionParameters(
+                    weight_name, node, weight_port_id, weight.size, reduction_axis
+                )
+                all_weight_params.append(weight_params)
+                weight_names.add(weight_name)
+
+        ratio_defining_params = self._get_ratio_defining_params(all_weight_params, is_last_layer_shared)
+        self._set_weight_compression_config(ratio_defining_params, model, graph, activations)
+        nncf_logger.info(self._get_bitwidth_distribution_str(all_weight_params, ratio_defining_params))
+
+        # Compress model using weight compression parameters
+        transformed_model = self._backend_entity.transform_model(
+            model, graph, track(all_weight_params, description="Applying Weight Compression")
+        )
+
+        self._backend_entity.dump_parameters(
             model,
-            nodes_to_compress,
-            self._mode,
-            self._ratio,
-            self._group_size,
-            self._all_layers,
-            activations,
-            self._sensitivity_metric,
+            parameters={
+                "mode": self._mode.value,
+                "group_size": self._group_size,
+                "ratio": self._ratio,
+                "all_layers": self._all_layers,
+                "ignored_scope": self._ignored_scope,
```

Comment:
What about checking for the empty ignored scope and dumping when it has some values only?
IMHO, empty ignored scope adds a noise
![image](https://github.com/openvinotoolkit/nncf/assets/4014476/36233218-b918-447f-afa4-2b93470de8f5)

Reply:
Thanks for your suggestion. This behavior corresponds to ignore-scoped methods such as nncf.quantize(). I think this needs to be changed for all methods with ignored scope at once. Please open a ticket to take action on your suggestion.

Reply:
opened ticket 129593